src/core/src/core_logic/ExecutionConfig.py

```diff
@@ -41,6 +41,7 @@ def __init__(self, env_layer, composite_logger, execution_parameters):
         self.log_folder = self.environment_settings[Constants.EnvSettings.LOG_FOLDER]
         self.config_folder = self.environment_settings[Constants.EnvSettings.CONFIG_FOLDER]
         self.status_folder = self.environment_settings[Constants.EnvSettings.STATUS_FOLDER]
+        self.events_folder = self.environment_settings[Constants.EnvSettings.EVENTS_FOLDER]
```

Comment:
Events folder is not guaranteed to be there in environment settings. -- note to self for looping back in review.

Please include tasks for a test plan for testing on an agent that doesn't support it and testing on an agent that supports it.

Reply:
Testing with 2 agent version, one with telemetry support and one without